### PR TITLE
arm64: dts: qcom: xiaomi-tulip: enable touchscreen input and bluetooth

### PR DESCRIPTION
**Pull request by M0Rf30**
As in the title

dmesg related lines

```
[25.060439] input: Novatek NT36XXX Touchscreen as /devices/platform/soc/c175000.i2c/i2c-0/0-0062/input/input2
```

related event device node is
`/dev/input/event2`

15/07
Hereby I also:
- add panel and descriptors in mainline driver
- add ncp and ldo voltage regulators as in downstream
- some little fixes to align with upstream naming
- dfps region
- disable dsi1 nodes (as in downstream, cause enabled in parent dtsi)
- remove non-working wifi and add only modem node.

22/07
- bluetooth

### CODE REVIEW

**Comment by M0Rf30** [comment]
@minlexx @Aboothahir I also noticed that other devices using the nt36525 use an unsupported `vcc-supply`

driver only supports optional power supplies named as vdd and vio (jasmine, lavender,  platina, ....)

**Comment by minlexx** [comment]
> @minlexx @Aboothahir I also noticed that other devices using the nt36525 use an unsupported `vcc-supply`
> 
> driver only supports optional power supplies named as vdd and vio (jasmine, lavender, platina, ....)

Do you mean that other devices got it wrong or what?

File: arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts [hunk 1]
```diff
@@ -518,6 +532,18 @@
...
+		pins = "gpio67";
+		drive-strength = <8>;
+		bias-pull-up;
+	};
+
+	ts_reset_active: ts-reset-active-state {
+		pins = "gpio66";
+		drive-strength = <8>;
+		bias-pull-up;
+	};
```
Comment (minlexx) [inline arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts]: `function = ` ? `gpio` ? can you link to your downstream source or where did you get that pinconf info from

File: arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts [hunk 2]
```diff
@@ -195,6 +195,20 @@
 
 &blsp_i2c1 {
 	status = "okay";
+
+	touchscreen: novatek@62 {
```
Comment (minlexx) [inline arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts]: The node should be named "touchscreen" and label is not needed (you don't reference `&touchscreen` anywhere, do you?)

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/writing-bindings.rst#n24

> use node names matching the class of the device

**Review by minlexx — CHANGES_REQUESTED** [review]

**Comment by M0Rf30** [comment]
> > @minlexx @Aboothahir I also noticed that other devices using the nt36525 use an unsupported `vcc-supply`
> > driver only supports optional power supplies named as vdd and vio (jasmine, lavender, platina, ....)
> 
> Do you mean that other devices got it wrong or what?

Yes, they got it wrong as you can see here
https://github.com/sdm660-mainline/linux/blob/sdm660-next-stable/drivers/input/touchscreen/nt36xxx.c#L712

in their dmesg, right now they should have
```
nt36xxx_ts 0-0062: supply vdd not found, using dummy regulator
nt36xxx_ts 0-0062: supply vio not found, using dummy regulator
```
simply because they are using vcc-supply (instead of vdd or vio)

File: arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts (hunk 1 above)
Comment (M0Rf30) [reply arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts]: [tulip.dts.txt](https://github.com/sdm660-mainline/linux/files/11644907/tulip.dts.txt)
in the file above you'll find
```
			ts_mux {

				ts_active {
					phandle = <0x17c>;

					mux {
						pins = "gpio66\0gpio67";
						function = "gpio";
					};

					config {
						pins = "gpio66\0gpio67";
						drive-strength = <0x10>;
						bias-pull-up;
					};
				};

				ts_reset_suspend {
					phandle = <0x17e>;

					mux {
						pins = "gpio66";
						function = "gpio";
					};

					config {
						pins = "gpio66";
						drive-strength = <0x02>;
						bias-pull-down;
					};
				};

				ts_int_suspend {
					phandle = <0x17d>;

					mux {
						pins = "gpio67";
						function = "gpio";
					};

					config {
						pins = "gpio67";
						drive-strength = <0x02>;
						bias-disable;
					};
				};
			};
```

Comment (minlexx) [reply arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts]: Then this definition should have `fuinction = "gpio";` otherwise it's no point of setting electrical properties of the pin. Overall conversion from downstream sould look like
```dts
	touchscreen@62 {
		...
		pinctrl-names = "default", "sleep";
		pinctrl-0 = <&ts_default>;
		pinctrl-1 = <&ts_int_sleep>, <&ts_reset_sleep>;
	};
...
...
&tlmm {
	ts_default: ts-pins-default-state {
		pins = "gpio66", "gpio67";
		function = "gpio";
		drive-strength = <16>;
		bias-pull-up;
	};

	ts_reset_sleep: ts-reset-sleep-state {
		pins = "gpio66";
		function = "gpio";
		drive-strength = <2>;
		bias-pull-down;
	};

	ts_int_sleep: ts-int-sleep-state {
		pins = "gpio67";
		function = "gpio";
		drive-strength = <2>;
		bias-disable;
	};
};
```

Comment (M0Rf30) [reply arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts]: Done and re-tested on the device

**Review by minlexx — APPROVED** [review]

File: arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts (hunk 2 above)
Comment (minlexx) [reply arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts]:
```diff
-touchscreen: novatek@62 {
+touchscreen@62 {
```

**Comment by M0Rf30** [comment]
@minlexx I've done: 
- general cleanup
- add resin (vol-)
- enable bluetooth
- enable touchscreen

nothing to do with wifi and panel enablement

**Comment by M0Rf30** [comment]
[dmesg-tulip-6.6.txt](https://github.com/sdm660-mainline/linux/files/13468033/dmesg-tulip-6.6.txt)

File: arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts [hunk 3]
```diff
@@ -174,7 +110,7 @@
 &blsp_i2c1 {
 	status = "okay";
 
-	touchscreen@62 {
+	touchscreen: novatek@62 {
```
Comment (minlexx) [inline arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts]: This is still not resolved. I do not understand why you changed it back after I fixed this line in 6.6 branch for you.

To be clear on terminology, naming of things in device tree is:
```
node_label: node_name@address { ... };
```
Here, your node_label (`touchscreen`) is not needed because it is never referenced anywhere and only takes more space in resulting compiled DTB file.

`node_name` should describe generic class of device, in this case `novatek` is not a class of device, `touchscreen` is.

Please change this line back to how it was: remove unused label and change name to be generic, `touchscreen@62`

File: arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts [hunk 4]
```diff
@@ -631,24 +503,13 @@
...
 
-&venus {
 	status = "okay";
+};
 
+&venus {
 	firmware-name = "qcom/venus-4.4/venus.mdt";
-};
 
-&wifi {
```
Comment (minlexx) [inline arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts]: Why did you remove wi-fi node? Even if it is not fully working, we can still have it here, I think.

I also think you're using kernel config which is different from mine, can I see your defconfig? (Run `make savedefconfig`and share defconfig file somewhere? To be precisefull command will look something like `make ARCH=arm64 CROSS_COMPILE=aarch64-alpine-linux-musl- savedefconfig`)

File: arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts (hunk 3 above)
Comment (M0Rf30) [reply arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts]: got confused by lavender dts, where also a label is reported (I always check also in comparison with other DTSes)

Comment (minlexx) [reply arch/arm64/boot/dts/qcom/sdm636-xiaomi-tulip.dts]: 🙈  I'm fixing this myself in 6.6-wip branch... https://github.com/sdm660-mainline/linux/commit/b4c33d266df108412aed9ae8040b497ca319bf95

**Comment by minlexx** [comment]
I've picked the commit https://github.com/M0Rf30/linux/commit/2ca7fcb334cabf4b4f974947a0e92d86366d36c8 to qcom-sdm660-6.6.y and squashed it with commit adding initial support for tulip. Since device dts is not upstream, it makes little sense to have separate millions commits that fix each other, because during upstreaming you'll need single commit (there should not be commits that change previous commits in single patchset scope). It now looks like this 5bbbd17630a89f89581fa1a1d9c75e192f91d201 . Likely more force pushes are incoming, since I see now that pm660_rradc and remoteproc_mss are specified twice :see_no_evil: ( fixup! ace6397b209cc05ecf5b36103382a856538be76c ) and I still need to do something with xiaomi-clover...

P.S. nt36672a panel support for tulip is still separate commit
